### PR TITLE
Promotion from AMD internal branch for 2022.Q2.3

### PR DESCRIPTION
**Pull request by WenqingLiAMD**
@kuhar

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,13 +41,23 @@ add_library(vkgc_headers INTERFACE)
...
 endif()
+
 if (LLPC_CLIENT_INTERFACE_MAJOR_VERSION)
     target_compile_definitions(vkgc_headers INTERFACE LLPC_CLIENT_INTERFACE_MAJOR_VERSION=${LLPC_CLIENT_INTERFACE_MAJOR_VERSION})
+else()
+    # LLPC is not compiled, so fall back to the latest version
+    target_compile_definitions(vkgc_headers INTERFACE LLPC_CLIENT_INTERFACE_MAJOR_VERSION=LLPC_INTERFACE_MAJOR_VERSION)
 endif()
 
+#if VKI_KHR_FRAGMENT_SHADER_BARYCENTRIC
```
Comment (kuhar) [inline CMakeLists.txt:56]: Do we need these comments?

File: CMakeLists.txt [hunk 2]
```diff
@@ -41,13 +41,23 @@ add_library(vkgc_headers INTERFACE)
 
 ### Options that affect the headers ####################################################################################
 option(LLPC_ENABLE_SHADER_CACHE "Enable experimental shader cache" OFF)
+
 if(LLPC_ENABLE_SHADER_CACHE)
-    target_compile_definitions(vkgc_headers INTERFACE LLPC_ENABLE_SHADER_CACHE=1)
+    target_compile_definitions(vkgc_headers INTERFACE LLPC_ENABLE_SHADER_CACHE)
```
Comment (kuhar) [inline CMakeLists.txt:46]: Why was this changed? IIRC we have some checks that expect this to be either 0 or 1, no? https://github.com/GPUOpen-Drivers/llpc/search?q=LLPC_ENABLE_SHADER_CACHE

File: tool/vfx/vfxError.h [hunk 3]
```diff
@@ -47,7 +47,7 @@
     int pos = vfxSnprintf(errorBuf, 4096, "Parse error at line %u: ", lineNum);                                        \
     pos += vfxSnprintf(errorBuf + pos, 4096 - pos, __VA_ARGS__);                                                       \
     pos += vfxSnprintf(errorBuf + pos, 4096 - pos, "\n");                                                              \
-    VFX_ASSERT(pos < 4096);                                                                                            \
+    VFX_NEVER_CALLED();                                                                                                \
```
Comment (kuhar) [inline tool/vfx/vfxError.h:50]: Is this intentional?

File: CMakeLists.txt (hunk 2 above)
Comment (WenqingLiAMD) [reply CMakeLists.txt:46]: @Flakebi could you please help comment?

**Comment by amdvlk-admin** [comment]
### Test summary for commit ba9ff4623ff7fd9ef052057aeb7fc4774a3189ef
<details><summary>CTS tests (Failed: 0/186371)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29989/55953 (53.6%)</li>
<li>Failed: 0/55953 (0.0%)</li>
<li>Not Supported: 25964/55953 (46.4%)</li><li>Warnings: 0/55953 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>

File: CMakeLists.txt (hunk 2 above)
Comment (Flakebi) [reply CMakeLists.txt:46]: As far as I can see, none of the uses checks for `=1`, so I made it consistent with the other flags that are not explicitly set to 1 either.
(The `config.llpc_enable_shader_cache == 'ON'` checks the CMake variable, not the preprocessor definition as far as I understand.)

File: llpc/CMakeLists.txt [hunk 4]
```diff
@@ -384,6 +370,7 @@ add_dependencies(amdllpc llpc_standalone_compiler)
 target_link_libraries(amdllpc PRIVATE llpc_standalone_compiler)
 set_compiler_options(amdllpc ${LLPC_ENABLE_WERROR})
 add_compile_definitions(amdllpc PRIVATE SH_EXPORTING)
+target_link_libraries(amdllpc PRIVATE spvgen_static)
```
Comment (Flakebi) [inline llpc/CMakeLists.txt:373]: This line seems to be added by mistake. It was removed in https://github.com/GPUOpen-Drivers/llpc/commit/b750345#diff-7b9725e04fc088aab06b9273d5399b456c4bdb52d536c1869ae16be6801e4db3
Looks like that part of the diff did not make it to the internal branch.

File: CMakeLists.txt (hunk 1 above)
Comment (WenqingLiAMD) [reply CMakeLists.txt:56]: Sorry, these code should not be here, I'll update and remove them.

File: llpc/CMakeLists.txt (hunk 4 above)
Comment (WenqingLiAMD) [reply llpc/CMakeLists.txt:373]: > This line seems to be added by mistake. It was removed in [b750345#diff-7b9725e04fc088aab06b9273d5399b456c4bdb52d536c1869ae16be6801e4db3](https://github.com/GPUOpen-Drivers/llpc/commit/b750345#diff-7b9725e04fc088aab06b9273d5399b456c4bdb52d536c1869ae16be6801e4db3) Looks like that part of the diff did not make it to the internal branch.

Thanks, will remove it.

**Comment by amdvlk-admin** [comment]
### Test summary for commit eb668587f7acc31ff0d8fe0bcd6f078d3fd99b3f
<details><summary>CTS tests (Failed: 1/185502)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 36631/65208 (56.2%)</li>
<li>Failed: 0/65208 (0.0%)</li>
<li>Not Supported: 28577/65208 (43.8%)</li><li>Warnings: 0/65208 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29119/55084 (52.9%)</li>
<li><details><summary>Failed: 1/55084 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.reconvergence.subgroup_uniform_control_flow_elect.compute.nesting4.0.45
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 25964/55084 (47.1%)</li><li>Warnings: 0/55084 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 38930/65210 (59.7%)</li>
<li>Failed: 0/65210 (0.0%)</li>
<li>Not Supported: 26280/65210 (40.3%)</li><li>Warnings: 0/65210 (0.0%)</li>
</ul></details>


</ul></details>
